src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -327,6 +327,14 @@ private static bool CastRemovalCouldCauseSignExtensionWarning(ExpressionSyntax c
             if (isNullLiteralCast && !originalConvertedType.IsReferenceType && !originalConvertedType.IsNullable())
                 return false;
 
+            // SomeType s = (Action)(() => {});  // Where there's a user defined conversion from Action->SomeType.
+            //
+            // This cast is necessary.  The language does not allow lambdas to be directly converted to the destination
+            // type without explicitly stating the intermediary reified delegate type.
+            var isAnonymousFunctionCast = castedExpressionNode.WalkDownParentheses() is AnonymousFunctionExpressionSyntax;
```

Comment:
>WalkDownParentheses [](http://example.com/codeflow?start=63&length=19)

assume it's not expensive to call as it was called a couple lines above too?

Reply:
Should be nearly free.  It's normally a no-op, or occasionally a walk of one node.